app/controllers/admin/suppliers/delivery_partners_controller.rb

```diff
@@ -55,17 +55,51 @@ def delivery_partner_success
         @delivery_partner = DeliveryPartner.find(params[:delivery_partner])
       end
 
-      def show_users; end
+      # endregion
+
+      def show
+        authorize @delivery_partner
+      end
+
+      def delete
+        authorize @delivery_partner, :destroy?
+      end
 
-      def show_lps; end
+      def edit
+        authorize @delivery_partner
 
-      def show_schools; end
+        @delivery_partner_form = DeliveryPartnerForm.new(
+          name: @delivery_partner.name,
+          lead_providers: @delivery_partner.lead_providers.map(&:id),
+          provider_relationship_hashes: @delivery_partner.provider_relationships.map do |relationship|
```

Comment:
One option would be to put the helper method to provider relationships - as such you would have all variables to create a string

Reply:
Not when we build it in the view